rest/src/main/java/io/syndesis/rest/v1/handler/setup/OAuthAppHandler.java

```diff
@@ -0,0 +1,141 @@
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeSet;
+
+/**
+ * This rest endpoint handles working with global oauth settings.
+ */
+@Path("/setup/oauth-apps")
+@Api(value = "oauth-apps")
+@Component
+public class OAuthAppHandler {
+
+    private final DataManager dataMgr;
+
+    public OAuthAppHandler(DataManager dataMgr) {
+        this.dataMgr = dataMgr;
+    }
+
+    // Since this a a view model DTO, and not a domain model lets define it here instead
+    // of placing it into the model module.
+    @SuppressFBWarnings(
+        value = "URF_UNREAD_PUBLIC_OR_PROTECTED_FIELD",
+        justification = "All fields are encode by jackson and sent to the UI")
+    public static final class OAuthApp {
+        public String id;
+        public String name;
+        public String icon;
+        public String clientId;
+        public String clientSecret;
+    }
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path(value = "")
+    public List<OAuthApp> get() {
+        ArrayList<OAuthApp> apps = new ArrayList<>();
+
+        List<Connector> items = dataMgr.fetchAll(Connector.class).getItems();
+        items.forEach(connector -> {
+            if (isOauthConnector(connector)) {
+                OAuthApp app = new OAuthApp();
+                app.id = connector.getId().get();
+                app.name = connector.getName();
+                app.icon = connector.getIcon();
+                app.clientId = getPropertyTaggedAs(connector, "oauth-client-id");
+                app.clientSecret = getPropertyTaggedAs(connector, "oauth-client-secret");
+                apps.add(app);
+            }
+        });
+
+        return apps;
+    }
+
+    @PUT
+    @Path(value = "/{id}")
+    @Consumes("application/json")
+    public void update(@PathParam("id") String id, OAuthApp app) {
+        Connector connector = dataMgr.fetch(Connector.class, id);
+        if( connector==null ) {
+            throw new WebApplicationException(Response.Status.NOT_FOUND);
+        }
+
+        Map<String, String> configuredProperties =  new HashMap<>(connector.getConfiguredProperties());
+        setPropertyTaggedAs(connector, configuredProperties, "oauth-client-id", app.clientId);
+        setPropertyTaggedAs(connector, configuredProperties,"oauth-client-secret", app.clientSecret);
+
+        // Updated the configuredProperties
+        Connector.Builder builder = connector.builder();
+        builder.configuredProperties(configuredProperties);
+        connector = builder.build();
+
+        dataMgr.update(connector);
+    }
+
+    private boolean isOauthConnector(Connector connector) {
+        TreeSet EMPTY = new TreeSet();
+        return connector.getProperties().values().stream().anyMatch(x -> {
+                return x.getTags().orElse(EMPTY).contains("oauth-client-id");
+            }
+        );
+    }
+
+    private String getPropertyTaggedAs(Connector connector, String name) {
```

Comment:
This might be useful on the Connector interface also, I think it can be used in Credentials to further simplify the applicators (can be worked on in another PR)